### PR TITLE
Update calendar event UI

### PR DESCRIPTION
**Pull request by hwangange**
Screenshot 2024-06-27 at 12 45 11 PM

### CODE REVIEW

File: src/lib/utils.js [hunk 1]
```diff
@@ -18,4 +19,31 @@ export function formatDate(date) {
...
+export function stringToDate(dateString) {
+   // Split the input string into components
+   const parts = dateString.split('-');
+   const year = parseInt(parts[0], 10);
+   const month = parseInt(parts[1], 10) - 1; // Months are 0-based in JavaScript Date
+   const day = parseInt(parts[2], 10);
+ 
+   // Create a Date object in UTC
+   const date = new Date(year, month, day);
+   return date;
```
Comment (TalSQ) [inline src/lib/utils.js]: didn't we want to convert?

```suggestion
   return toUTCDate(date);
```

File: src/components/calendarpage.jsx [hunk 2]
```diff
@@ -194,9 +223,10 @@ export function CalendarPage() {
         selectable={true}
         plugins={[dayGridPlugin, interactionPlugin]}
         dateClick={handleDateClick}
-        eventClick={openModalFromCalendar}
+        eventClick={openModalFromCalendar} // TODO: Display a different modal
```
Comment (TalSQ) [inline src/components/calendarpage.jsx]: What's with the different modal? And do we need to do it before merging in this PR?

File: src/lib/utils.js (hunk 1 above)
Comment (hwangange) [reply src/lib/utils.js]: ah comment is misleading, should return the YYYY-MM-DD date at 00:00, user's time zone

File: src/components/calendarpage.jsx [hunk 3]
```diff
@@ -131,13 +139,34 @@ export function CalendarPage() {
...
+      <div>
+        {info.dayNumberText}
+      </div>
+    );
+  }
+
   const EventItem = ({ info }) => {
     const { event } = info;
     const flowType = event.extendedProps.flowType;
+    const date = toUTCDate(event._instance.range.start);
```
Comment (TalSQ) [inline src/components/calendarpage.jsx]: Note: I think the '_' is indicating that the instance is meant to be private.
Are you sure there isn't an accessor for this?

Not a blocker

File: src/components/calendarpage.jsx (hunk 2 above)
Comment (hwangange) [reply src/components/calendarpage.jsx]: Updated TODO: `TODO: Fix existing modal to support editing period data or display a different modal`

File: src/components/calendarpage.jsx [hunk 4]
```diff
@@ -131,13 +139,34 @@ export function CalendarPage() {
...
+    // Check if date overlaps with a period
+    for (let i = 0; i < periods.length; i++) {
+      const period = periods[i];
+      const dayCellDate = info.date;
+      const startDate = stringToDate(period.startDate);
+      const endDate = stringToDate(period.endDate);
+      if (dayCellDate >= startDate && dayCellDate <= endDate) {
+        return (<div></div>);
+      }
+    }
```
Comment (TalSQ) [inline src/components/calendarpage.jsx:152]:
```suggestion
     const dayCellDate = info.date;
     const isOverlap = periods.some(period => {
        const startDate = stringToDate(period.startDate);
        const endDate = stringToDate(period.endDate);
        return dayCellDate >= startDate && dayCellDate <= endDate;
      })
      if (isOverlap) {  
        return (<div></div>);
      }
    }
```

File: src/components/calendarpage.jsx (hunk 3 above)
Comment (hwangange) [reply src/components/calendarpage.jsx]: Fixed

File: src/components/calendarpage.jsx (hunk 4 above)
Comment (TalSQ) [reply src/components/calendarpage.jsx:152]: also might be good to have a function isInPeriod

```
function isInPeriod(aDate, aPeriod) {
        const startDate = stringToDate(period.startDate);
        const endDate = stringToDate(period.endDate);
        return aDate >= startDate && aDate <= endDate;
}

and then..
if (periods.some(period => isInPeriod(info.date, period))) {
  return (<div></div>);
}
```

The isOverlap would go really nicely on a Period object :)

File: src/components/calendarpage.jsx [hunk 5]
```diff
@@ -5,8 +5,8 @@ import { useEffect, useState } from "react";
 import CurrentPeriodControl from "@/components/current-period-control";
 import { PeriodDialog } from "@/components/period-dialog";
 import { v4 as uuidv4 } from 'uuid';
-import { formatDate } from "@/lib/utils";
 import PeriodTracker from '@/lib/dwn/period-tracker';
+import { formatDate, stringToDate, toUTCDate } from "@/lib/utils";
```
Comment (TalSQ) [inline src/components/calendarpage.jsx:9]: good move! :)

File: src/components/calendarpage.jsx (hunk 4 above)
Comment (hwangange) [reply src/components/calendarpage.jsx:152]: Getting an error that `.some` is not a function on `periods`
This is a hack week project, so I'm not super dedicated to improving the code if it works. but feel free to submit these improvements in a separate PR

**Review by TalSQ — APPROVED** [review]

**Review by blackgirlbytes — APPROVED** [review]

File: src/components/calendarpage.jsx (hunk 4 above)
Comment (TalSQ) [reply src/components/calendarpage.jsx:152]: Yup, mostly wanted to highlight that using builtin collection functions can help get rid of the `for loop` mechanics and draw more attention to what the for loop is doing :)

**Pull request merged**